sandbox/Test.scala

```diff
@@ -1,13 +1,15 @@
-import scalanative.native._
+import scalanative.native._, FunctionPtr._
```

Comment:
`js.ThisFunctionN` extends `js.ThisFunction`, and the implicit conversions are in the companion object of `js.ThisFunction`.

If you want automatic resolution, you either need `FunctionPtrN` to extend `FunctionPtr`, or put the implicit conversion to `FunctionPtrN` in the companion of `FunctionPtrN`.